vocab-builder-cli/src/main/java/com/github/tkurz/sesame/vocab/Main.java

```diff
@@ -100,38 +100,82 @@ public static void main(String[] args) {
             if (cli.hasOption('l')) {
                 builder.setPreferredLanguage(cli.getOptionValue('l'));
             }
-            if (cli.hasOption('S')) {
-                builder.setStringPropertySuffix(cli.getOptionValue('S'));
-            } else {
-                builder.setStringPropertySuffix(null);
-            }
-            if (cli.hasOption('P')) {
-                builder.setStringPropertyPrefix(cli.getOptionValue('P'));
-            } else {
-                builder.setStringPropertyPrefix(null);
-            }
+            CaseFormat caseFormat = null;
+            final GenerationSetting uriGenerationSetting = new GenerationSetting(); 
             if (cli.hasOption('c')) {
+            	String uriCaseFormat = cli.getOptionValue('c');
+            	uriGenerationSetting.setCaseFormat(uriCaseFormat);
                 try {
-                    final CaseFormat caseFormat = CaseFormat.valueOf(cli.getOptionValue('c'));
+                    caseFormat = CaseFormat.valueOf(cli.getOptionValue('c'));
                     if (caseFormat == null) {
                         throw new ParseException("Did not recognise constantCase: Must be one of " + Arrays.asList(CaseFormat.values()));
                     }
-                    builder.setConstantCase(caseFormat);
                 } catch (IllegalArgumentException e) {
                     throw new ParseException("Did not recognise constantCase: Must be one of " + Arrays.asList(CaseFormat.values()));
                 }
             }
-            if (cli.hasOption('C')) {
-                try {
-                    final CaseFormat caseFormat = CaseFormat.valueOf(cli.getOptionValue('C'));
-                    if (caseFormat == null) {
+            // in any case, do URI generation from the command line
+            builder.setUriGeneration(uriGenerationSetting);
+            
+            // options for string generation - when NO option present--> no string generation
+            if ( cli.hasOption('S') || cli.hasOption('P') || cli.hasOption('C')) {
+            	GenerationSetting setting = new GenerationSetting();
+            	if ( cli.hasOption('P')) {
+            		setting.setConstantPrefix(cli.getOptionValue('P'));
+            	}
+            	if ( cli.hasOption('S')) {
+            		setting.setConstantSuffix(cli.getOptionValue('S'));
+            	}
+                if (cli.hasOption('C')) {
+                	setting.setCaseFormat(cli.getOptionValue('C'));
+                    try {
+                        final CaseFormat strCaseFormat = CaseFormat.valueOf(cli.getOptionValue('C'));
+                        if (caseFormat == null) {
```

Comment:
This looks like it is checking the wrong variable, as a result of pushing the caseFormat variable into the main method scope.